<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of single cell RNA-Seq data using scTernary â€¢ scTernary</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of single cell RNA-Seq data using scTernary">
<meta property="og:description" content="scTernary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scTernary</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/scTernary.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jinming-cheng/scTernary/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of single cell RNA-Seq data using
scTernary</h1>
                        <h4 data-toc-skip class="author">Jinming Cheng,
Yunshun Chen and Gordon K. Smyth</h4>
            <address class="author_afil">
      Walter and Eliza Hall Institute of Medical Research, Melbourne,
Australia<br><h4 data-toc-skip class="date">2023-11-05</h4>
      
      
      <div class="hidden name"><code>scTernary.Rmd</code></div>

    </address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Ternary plot is a good method to visualize three-dimensional data.
scTernary is an easy and user friendly package to show three different
single cell populations as well as the potential intermediate cell
populations on the ternary plot.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The <em>scTernary</em> package can be installed from GitHub by
using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"jinming-cheng/scTernary"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ternary-plot-analysis-for-10x-single-cell-rna-seq-data">Ternary plot analysis for 10x single cell RNA-Seq data<a class="anchor" aria-label="anchor" href="#ternary-plot-analysis-for-10x-single-cell-rna-seq-data"></a>
</h2>
<p>Load package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scTernary</span><span class="op">)</span></span></code></pre></div>
<p>Obtain raw count matrix from an example <code>Seurat</code> object
containing cells from adult mouse mammary gland epithelium.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_exp_mat</span> <span class="op">&lt;-</span> <span class="va">example_seu</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span></span>
<span><span class="va">data_exp_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">data_exp_mat</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: Matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">data_exp_mat</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 13750    80</span></span></code></pre></div>
<p>Mouse signature genes for Basal, LP and ML cells</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">anno_signature_genes_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt;        GeneID        Symbol Chr gene_type</span></span>
<span><span class="co">#&gt; 497097 497097          Xkr4   1     Basal</span></span>
<span><span class="co">#&gt; 329093 329093          Cpa6   1     Basal</span></span>
<span><span class="co">#&gt; 320492 320492 A830018L16Rik   1     Basal</span></span>
<span><span class="co">#&gt; 240725 240725         Sulf1   1     Basal</span></span>
<span><span class="co">#&gt; 14048   14048          Eya1   1     Basal</span></span>
<span><span class="co">#&gt; 226866 226866        Sbspon   1     Basal</span></span></code></pre></div>
<p>Number of signature genes for each cell type</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">anno_signature_genes_mouse</span><span class="op">$</span><span class="va">gene_type</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Basal    LP    ML </span></span>
<span><span class="co">#&gt;  1470   428   529</span></span></code></pre></div>
<p>Prepare data for ternary plot. A constant raw count cut-off of 0 is
used to count the number of genes for each signature gene type.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_for_ternary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_data_for_ternary.html">generate_data_for_ternary</a></span><span class="op">(</span></span>
<span>  data_exp_mat <span class="op">=</span> <span class="va">data_exp_mat</span>,</span>
<span>  anno_signature_genes <span class="op">=</span> <span class="va">anno_signature_genes_mouse</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="st">"Symbol"</span>,</span>
<span>  gene_type_col <span class="op">=</span> <span class="st">"gene_type"</span>,</span>
<span>  weight_by_gene_count <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cutoff_exp <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  prior_count <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">data_for_ternary</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             Basal         LP         ML</span></span>
<span><span class="co">#&gt; Adult-CGGAGGCTGTCTGA-1 0.06108202 0.03217158 0.01113586</span></span>
<span><span class="co">#&gt; Adult-ATCTCAACGAACCT-1 0.13263525 0.04825737 0.03563474</span></span>
<span><span class="co">#&gt; Adult-TGGAAGCTTACAGC-1 0.09773124 0.05361930 0.02449889</span></span>
<span><span class="co">#&gt; Adult-CTTCTAGACTGTGA-1 0.07329843 0.03753351 0.02672606</span></span>
<span><span class="co">#&gt; Adult-AAGACAGATTGGTG-1 0.06806283 0.02949062 0.01336303</span></span>
<span><span class="co">#&gt; Adult-TTGAGGACGTTTGG-1 0.06806283 0.08579088 0.01559020</span></span></code></pre></div>
<p>Add metadata to the data for ternary plot</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_for_ternary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">data_for_ternary</span>,<span class="va">example_seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<p>Draw the ternary plot colored by cell type. Mouse mammary gland
epithelium contains three major cell types: Basal, LP and ML. In this
example, an additional luminal intermediate cell population can be
easily observed by using the ternary plot method.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">scTernary</span><span class="fu">::</span><span class="fu"><a href="../reference/vcdTernaryPlot_functions.html">vcdTernaryPlot</a></span><span class="op">(</span><span class="va">data_for_ternary</span>,</span>
<span>                          order_colnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                          point_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                          group <span class="op">=</span> <span class="va">data_for_ternary</span><span class="op">$</span><span class="va">cell_type_label</span>,</span>
<span>                          show_legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                          scale_legend <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                          legend_position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>,<span class="fl">0.5</span><span class="op">)</span>,</span>
<span>                          legend_vertical_space <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                          legend_text_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="scTernary_files/figure-html/unnamed-chunk-8-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="ternary-plot-analysis-for-fluidigm-c1-data-or-bulk-rna-seq-data">Ternary plot analysis for Fluidigm C1 data or bulk RNA-Seq data<a class="anchor" aria-label="anchor" href="#ternary-plot-analysis-for-fluidigm-c1-data-or-bulk-rna-seq-data"></a>
</h2>
<div class="section level3">
<h3 id="ternary-plot-analysis-using-a-constant-cut-off">Ternary plot analysis using a constant cut-off<a class="anchor" aria-label="anchor" href="#ternary-plot-analysis-using-a-constant-cut-off"></a>
</h3>
<p>The number of genes and library size of Fluidigm C1 single cell
RNA-Seq data is quite similar to bulk RNA-Seq data. Hence, we use a
small bulk RNA-Seq dataset to show the ternary plot analysis for
Fluidigm C1 data.</p>
<p>Example bulk RNA-seq data (using CPM)</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cpm</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">example_dge_data</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<p>Three samples for each group</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_dge_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span></span>
<span><span class="co">#&gt; [1] LP    ML    Basal Basal ML    LP    Basal ML    LP   </span></span>
<span><span class="co">#&gt; Levels: Basal LP ML</span></span></code></pre></div>
<p>Generate data for ternary plot using a CPM cut-off of 100</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_for_ternary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_data_for_ternary.html">generate_data_for_ternary</a></span><span class="op">(</span></span>
<span>  data_exp_mat <span class="op">=</span> <span class="va">cpm</span>,</span>
<span>  anno_signature_genes <span class="op">=</span> <span class="va">anno_signature_genes_mouse</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  gene_type_col <span class="op">=</span> <span class="st">"gene_type"</span>,</span>
<span>  weight_by_gene_count <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cutoff_exp <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  prior_count <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">data_for_ternary</span><span class="op">)</span></span>
<span><span class="co">#&gt;                Basal         LP         ML</span></span>
<span><span class="co">#&gt; 10_6_5_11 0.02244898 0.34579439 0.07750473</span></span>
<span><span class="co">#&gt; 9_6_5_11  0.02040816 0.16822430 0.29300567</span></span>
<span><span class="co">#&gt; purep53   0.21020408 0.06074766 0.03402647</span></span>
<span><span class="co">#&gt; JMS8-2    0.21496599 0.05841121 0.03591682</span></span>
<span><span class="co">#&gt; JMS8-3    0.02380952 0.21495327 0.26654064</span></span>
<span><span class="co">#&gt; JMS8-4    0.02040816 0.33878505 0.09451796</span></span></code></pre></div>
<p>Drawing the ternary plot</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/vcdTernaryPlot_functions.html">vcdTernaryPlot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data_for_ternary</span>,</span>
<span>  order_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>  group <span class="op">=</span> <span class="va">example_dge_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span>,</span>
<span>  group_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span><span class="op">)</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  show_legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  legend_point_size <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  legend_position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>,<span class="fl">0.5</span><span class="op">)</span>,</span>
<span>  scale_legend <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="scTernary_files/figure-html/TernaryPlotConstandCutoff-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="ternary-plot-analysis-using-estimated-cut-offs">Ternary plot analysis using estimated cut-offs<a class="anchor" aria-label="anchor" href="#ternary-plot-analysis-using-estimated-cut-offs"></a>
</h3>
<p>We can also find the optimized cut-off for each sample (or cell) to
calculate the signature gene proportion for the sample (or cell). The
optimized cut-off is estimated by maximizing the coefficient of
variation (CV) of the signature gene proportion by a grid search
method.</p>
<p>Example bulk RNA-seq data (using logCPM)</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lcpm</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">example_dge_data</span><span class="op">$</span><span class="va">counts</span>,log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Three samples for each group</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_dge_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span></span>
<span><span class="co">#&gt; [1] LP    ML    Basal Basal ML    LP    Basal ML    LP   </span></span>
<span><span class="co">#&gt; Levels: Basal LP ML</span></span></code></pre></div>
<p>Estimate a optimized cut-off for each sample (or cell)</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estimated_cutoffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimate_optimized_cutoffs.html">estimate_optimized_cutoffs</a></span><span class="op">(</span></span>
<span>  data_exp_mat <span class="op">=</span> <span class="va">lcpm</span>,</span>
<span>  anno_signature_genes <span class="op">=</span> <span class="va">anno_signature_genes_mouse</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  gene_type_col <span class="op">=</span> <span class="st">"gene_type"</span>,</span>
<span>  weight_by_gene_count <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  prior_count <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  do_parallel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_cores <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Generate data for ternary plot</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_for_ternary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_data_for_ternary.html">generate_data_for_ternary</a></span><span class="op">(</span></span>
<span>  data_exp_mat <span class="op">=</span> <span class="va">lcpm</span>,</span>
<span>  anno_signature_genes <span class="op">=</span> <span class="va">anno_signature_genes_mouse</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  gene_type_col <span class="op">=</span> <span class="st">"gene_type"</span>,</span>
<span>  weight_by_gene_count <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cutoff_exp <span class="op">=</span> <span class="va">estimated_cutoffs</span>,</span>
<span>  prior_count <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">data_for_ternary</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 Basal          LP          ML</span></span>
<span><span class="co">#&gt; 10_6_5_11 0.001360544 0.091121495 0.003780718</span></span>
<span><span class="co">#&gt; 9_6_5_11  0.001360544 0.004672897 0.018903592</span></span>
<span><span class="co">#&gt; purep53   0.100000000 0.025700935 0.009451796</span></span>
<span><span class="co">#&gt; JMS8-2    0.031292517 0.004672897 0.003780718</span></span>
<span><span class="co">#&gt; JMS8-3    0.001360544 0.011682243 0.003780718</span></span>
<span><span class="co">#&gt; JMS8-4    0.001360544 0.063084112 0.003780718</span></span></code></pre></div>
<p>Drawing the ternary plot</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/vcdTernaryPlot_functions.html">vcdTernaryPlot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data_for_ternary</span>,</span>
<span>  order_colnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>  group <span class="op">=</span> <span class="va">example_dge_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span>,</span>
<span>  group_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"green"</span>,<span class="st">"blue"</span><span class="op">)</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  show_legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  legend_point_size <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  legend_position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>,<span class="fl">0.5</span><span class="op">)</span>,</span>
<span>  scale_legend <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="scTernary_files/figure-html/TernaryPlotOptimizedCutoff-1.png" width="100%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jinming Cheng.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
